### ISSUE TITLE
article template not knitting to pdf

### ISSUE THREAD

**Issue by casperhart** [rjournal/rjtools#43]
Hi, I seem to have trouble knitting the r journal article template `quokka-bilby.Rmd` to pdf. The steps I use to knit are:

1. In an empty directory, I use `rjtools::create_article()` to create a blank template
2. run `rmarkdown::render("rjarticle/quokka-bilby.Rmd")` without modifying the template at all

The result is that the HTML document is created, but then the PDF gives error:

```
Output created: quokka-bilby.html
Error: callr subprocess failed: argument is of length zero
Type .Last.error.trace to see where the error occurred
```

Typing `.Last.error.trace` gives:

```
r$> .Last.error.trace

 Stack trace:

 Process 93847:
 1. rmarkdown::render("rjarticle/quokka-bilby.Rmd")
 2. output_format$on_exit()
 3. rjtools:::overlay()
 4. callr::r(function(input) { ...
 5. callr:::get_result(output = out, options)
 6. throw(newerr, parent = remerr[[2]])

 x callr subprocess failed: argument is of length zero

 Process 94024:
 18. (function (input)  ...
 19. rmarkdown::render(input, output_format = "rjtools::rjournal_pdf_article")
 20. output_format$post_processor(front_matter, input, output_file,  ...
 21. base:::.handleSimpleError(function (e)  ...
 22. h(simpleError(msg, call))

 x argument is of length zero
```

Some more information:

- I'm using R version 4.2.0
- I have installed tex through `tinytex::install_tinytex()`, and can knit other documents using the `pdf_document` format.

Any help would be much appreciated.

Thanks,
Casper

**Comment by dicook** [comment]
I am getting an error too, with the same error message as you. Looks like a fix is needed.

> On 30 Jun 2022, at 3:46 pm, casperhart ***@***.***> wrote:
> 
> 
> Hi, I seem to have trouble knitting the r journal article template quokka-bilby.Rmd to pdf. The steps I use to knit are:
> 
> 	• In an empty directory, I use rjtools::create_article() to create a blank template
> 	• run rmarkdown::render("rjarticle/quokka-bilby.Rmd") without modifying the template at all
> The result is that the HTML document is created, but then the PDF gives error:
> 
> Output created: quokka-bilby.html
> Error: callr subprocess failed: argument is of length zero
> Type .Last.error.trace to see where the error occurred
> 
> Typing .Last.error.trace gives:
> 
> r$> .Last.error.trace
> 
>  Stack trace:
> 
>  Process 93847:
>  1. rmarkdown::render("rjarticle/quokka-bilby.Rmd")
>  2. output_format$on_exit()
>  3. rjtools:::overlay()
>  4. callr::r(function(input) { ...
>  5. callr:::get_result(output = out, options)
>  6. throw(newerr, parent = remerr[[2]])
> 
>  x callr subprocess failed: argument is of length zero
> 
>  Process 94024:
>  18. (function (input)  ...
>  19. rmarkdown::render(input, output_format = "rjtools::rjournal_pdf_article")
>  20. output_format$post_processor(front_matter, input, output_file,  ...
>  21. base:::.handleSimpleError(function (e)  ...
>  22. h(simpleError(msg, call))
> 
>  x argument is of length zero
> 
> Some more information:
> 
> 	• I'm using R version 4.2.0
> 	• I have installed tex through tinytex::install_tinytex(), and can knit other documents using the pdf_document format.
> Any help would be much appreciated.
> 
> Thanks,
> Casper
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.
> 

cheers,
Di

---
Dianne Cook
***@***.***

**Comment by huizezhang-sherry** [comment]
Hi Casper, 

Similar issue as you and Di with the lastest github version, but I can get the knit to work at ref: e38c3bd835e4a91e928cc4e4d4e5f7b2681edfa2.

So a temporary solution can be`remotes::install_github("rjournal/rjtools", ref = "e38c3bd835e4a91e928cc4e4d4e5f7b2681edfa2")`?

Regards,
Sherry

**Comment by mitchelloharawild** [comment]
Fixed in 7982b35bc68f24ec10d0e328efa5dc401a20d0ce, thanks!

**Comment by dicook** [comment]
It is the latest changes that Mitch has made to accommodate the issue build functionality. Hopefully it is a simple fix for him.

> On 30 Jun 2022, at 7:30 pm, Sherry Zhang ***@***.***> wrote:
> 
> 
> Hi Casper,
> 
> Similar issue as you and Di with the lastest github version, but I can get the knit to work at ref: e38c3bd.
> 
> So a temporary solution can beremotes::install_github("rjournal/rjtools", ref = "e38c3bd835e4a91e928cc4e4d4e5f7b2681edfa2")?
> 
> Regards,
> Sherry
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.
> 

cheers,
Di

---
Dianne Cook
***@***.***
